pkg/corerp/backend/deployment/deploymentprocessor.go

```diff
@@ -287,6 +287,18 @@ func (dp *deploymentProcessor) Deploy(ctx context.Context, id resources.ID, rend
 
 		err := dp.deployOutputResource(ctx, id, rendererOutput, computedValues, &handlers.PutOptions{Resource: &outputResource, DependencyProperties: deployedOutputResourceProperties})
 		if err != nil {
+			// add the last deployed resource  which was deployed with error to deployed list
+			// cleanup all deployed output resources in reverse order since app deployment has failed
+			outputResource := rpv1.OutputResource{
+				LocalID: outputResource.LocalID,
+				ID:      outputResource.ID,
+			}
+			deployedOutputResources = append(deployedOutputResources, outputResource)
+			deleteErr := dp.Delete(ctx, id, deployedOutputResources)
+			if deleteErr != nil {
+				logger.Info("Failed to cleanup deployed output resources for resource %q. Error: %s", id.String(), deleteErr.Error())
+			}
+
```

Comment:
WDYT would happen in the following scenario?

- User deploys an application successfully. They are using a Recipe to create a Redis cache, and a cloud-hosted Redis cache is created successfully.
- User updates the application. The recipe runs again, but this time it fails - let's say it was on outage on the cloud-provider's part.
- Then what ???

Reply:
I think it would unnecessarily delete the cache too, with potential data loss . :(
I am trying to fix a tricky scenario where rad app delete can't clean up resources upon deployment error since we dont track o/p resources at all in that case.. I am not sure it would be an improvement if I deleted just the errored outputresource in that case?

Reply:
> I think it would unnecessarily delete the cache too, with potential data loss . 

Yup, that's right 😆

Reply:
Instead of deleting resource and returning error, what if we can add `status` in deploymentOutput to track the failure state? For updating scenario (failed scenario), we can upsert to the outputresources to keep old resource references for deletion. WDYT?